### PR TITLE
`@init` → `@local` + `@set init = ...`

### PR DESCRIPTION
**Pull request by carstenbauer**
Closes https://github.com/JuliaFolds2/OhMyThreads.jl/issues/64

* [x] `@init` → `@local`
* [x] `@set init = ...`
* [x] drop deprecated `SpawnAllScheduler` (since we're preparing a breaking release here anyways)

### CODE REVIEW

**Comment by carstenbauer** [comment]
```julia
julia> macro local(args...)
       end
ERROR: ParseError:
# Error @ REPL[16]:1:7
macro local(args...)
#     └───┘ ── invalid macro name
Stacktrace:
 [1] top-level scope
   @ none:1
```
Of course, we only really define this as a proper macro for documentation reasons. Guess we have to change the name? (cc @MasonProtter )

**Comment by MasonProtter** [comment]
That's just a parsing thing we can work around like so:
```julia
julia> macro var"local"(args...)
           "boo!"
       end;

julia> @local 1 2 3
"boo!"
```

File: src/macros.jl [hunk 1]
```diff
@@ -99,14 +100,14 @@ end
...
         x::Vector{Int64} = rand(Int, 3)
         M::Matrix{Float64} = rand(3, 3)
     end
     # ...
 end
 ```
 """
-macro init(args...)
-    error("The @init macro may only be used inside of a @tasks block.")
+macro local(args...)
```
Comment (MasonProtter) [inline src/macros.jl:111]:
```suggestion
macro var"local"(args...)
```

**Comment by carstenbauer** [comment]
This trick [doesn't seem to work on 1.9](https://github.com/JuliaFolds2/OhMyThreads.jl/actions/runs/8140941184/job/22247143072?pr=67#step:6:92).

**Pull request merged**